src/lxml/_elementpath.py

```diff
@@ -83,11 +82,11 @@ def xpath_tokenizer(pattern, namespaces=None):
                 try:
                     if not namespaces:
                         raise KeyError
-                    yield ttype, "{%s}%s" % (namespaces[prefix], uri)
+                    yield ttype, f"{{{namespaces[prefix]}}}{uri}"
```

Comment:
I prefer not using f-strings in this specific case. The triple braces make the expression much more complex than the original one. And recent Python (and Cython) versions will convert this into an f-string anyway.

Reply:
I don't this the triple braces are needed here? I could manually adjust that.